commons/src/main/java/org/infinispan/commons/dataconversion/Encoder.java

```diff
@@ -0,0 +1,32 @@
+package org.infinispan.commons.dataconversion;
+
+/**
+ * Used to convert data between read/write format and storage format.
+ * @since 9.1
+ */
+public interface Encoder {
+
+   /**
+    * @param content Key in the read/write format.
+    * @return Object key in the storage format.
+    */
+   Object keyToStorage(Object content);
+
+   /**
+    * @param content value in the read/write format.
+    * @return Object value in the storage format.
+    */
+   Object valueToStorage(Object content);
+
+   Object keyFromStorage(Object content);
+
+   Object valueFromStorage(Object content);
+
+   /**
+    * @return if true, will perform stream and related operation in the storage format.
+    */
+   boolean isStorageFormatFilterable();
+
+   Object extractIndexable(Object content);
```

Comment:
Some additional javadoc on these methods would be much appreciated. Especially since this is an SPI it seems right?